msteams-platform/toolkit/add-single-sign-on.md

```diff
@@ -125,236 +127,295 @@ SSO command handler `ProfileSsoCommandHandler` uses an Azure AD token to call Mi
-                            contentUrl: photoUrl
-                        }
-                    ]
-                };
-                return photoMessage;
-            } else {
-                return "Could not retrieve your photo from Microsoft Graph. Please make sure you have uploaded your photo.";
-            }
-        }
-    }
+1. Add `PhotoSsoCommandHandler` instance to `ssoCommands` array in `bot/src/internal/initialize.ts`:
 
-    module.exports = {
-        PhotoSsoCommandHandler,
-    };
+   ```ts
+   // for TypeScript:
+   import { PhotoSsoCommandHandler } from "../photoSsoCommandHandler";
+
+   export const commandBot = new ConversationBot({
+       ...
+       command: {
+           ...
+           ssoCommands: [new ProfileSsoCommandHandler(), new PhotoSsoCommandHandler()],
+       },
+   });
+   ```
+
+   ```javascript
+   // for JavaScript:
+   ...
+   const { PhotoSsoCommandHandler } = require("../photoSsoCommandHandler");
+
+   const commandBot = new ConversationBot({
+       ...
+       command: {
+           ...
+           ssoCommands: [new ProfileSsoCommandHandler(), new PhotoSsoCommandHandler()]
+       },
+   });
+   ...
+
+   ```
 
-    ```
+1. Register your command in the Teams app manifest. Open `templates/appPackage/manifest.template.json`, and add following lines under `commands` in `commandLists` of your bot:
 
-1. Add `PhotoSsoCommandHandler` instance to `ssoCommands` array in `bot/src/internal/initialize.ts`:
+   ```JSON
 
-    ```ts
-    // for TypeScript:
-    import { PhotoSsoCommandHandler } from "../photoSsoCommandHandler";
-
-    export const commandBot = new ConversationBot({
-        ...
-        command: {
-            ...
-            ssoCommands: [new ProfileSsoCommandHandler(), new PhotoSsoCommandHandler()],
-        },
-    });
-    ```
+   {
+       "title": "photo",
+       "description": "Show user photo using Single Sign On feature"
+   }
+
+   ```
+
+</details>
+
+<details>
+<summary><b>Message extension project
+</b></summary>
 
-    ```javascript
-    // for JavaScript:
-    ...
-    const { PhotoSsoCommandHandler } = require("../photoSsoCommandHandler");
+The sample business logic provides a handler `TeamsBot` extends TeamsActivityHandler and override `handleTeamsMessagingExtensionQuery`. 
 
-    const commandBot = new ConversationBot({
-        ...
-        command: {
-            ...
-            ssoCommands: [new ProfileSsoCommandHandler(), new PhotoSsoCommandHandler()]
-        },
-    });
-    ...
+You can update the query logic in the `handleMessageExtensionQueryWithToken` with token which is obtained by using the logged-in Teams user token.
 
+To make this work in your application:
+1. Move the `auth/bot/public` folder to `bot`. This folder contains HTML pages that the bot application hosts. When single sign-on flows are initiated with AAD, AAD will redirect the user to these pages.
+
+1. Modify your `bot/index` to add the appropriate `restify` routes to these pages.
```

Comment:
As per guidelines to write a folder path: In command syntax, use path to represent only the folder portion of the full path: copy [drive:][path]filename.

Request you to check and update the file path across the doc.

Reply:
As discussed with the PM, the PM explained that we can't add a dive path as it depends on where the user installs the package. He also confirmed that the folder structure remains the same.

Change not required.